### DISCUSSION TITLE
Ideas for Blockchain Roadmap

### DISCUSSION THREAD

**Discussion by kaiwalyakoparkar** [commclassroom/roadmaps#137] [Ideas]
Please add your ideas about blockchain roadmap below

**Comment by awesamarth** [comment]
I think Foundry and Hardhat should be added to the list of Tools and also to the list of Open Source projects that we can contribute to.

**Comment by shoaibshebi** [comment]
@kaiwalyakoparkar 
I am preparing for a PR which is about roadmap to the Private Blockchain as we discussed on twitter, hopefully, would open tomorrow night.

Thanks

**Comment by Instidevfut** [comment]
> Please add your ideas about the blockchain roadmap below

Hi @kaiwalyakoparkar, attaching all the links about the current infrastructure for Private & Enterprise Blockchain 

https://www.hyperledger.org/use/fabric
https://www.corda.net/
https://consensys.net/quorum/
https://docs.kaleido.io/ (semi- code platform for enterprise blockchain solutions)
https://hyperledger.github.io/firefly/
https://geth.ethereum.org/ 

I would love to create more detailed diagrammatic resources for the community to get more clarity, as there is not much content available in a structured manner just I don't what community classroom follows the process to create that.

Thank you

**Comment by shoaibshebi** [comment]
Hi @kaiwalyakoparkar @kunal-kushwaha 

Here I've designed a rough roadmap for Private Blockchain. Have a look.
Its even less than the half of the complete road map that is in my mind.

![Private Blockchain drawio(1)](https://user-images.githubusercontent.com/47749364/185802010-be902d76-ab8a-48a8-a4e0-9f47dd22e8ad.png)

But i could'nt find the repo for https://www.commclassroom.org/roadmaps, where i can open the PR after completing the roadmap.
I think https://www.commclassroom.org/roadmaps is not available repo for openly contribute but only to put the missing things in resources.
Do guide me , if 'm wrong.

Thanks

**Reply by kaiwalyakoparkar** [reply]
Please send the resouces in .md format

**Reply by kaiwalyakoparkar** [reply]
Maybe create a github gist for it

**Reply by cosmicqbit** [reply]
@shoaibshebi The roadmaps you see on the [official roadmaps](https://www.commclassroom.org/roadmaps) page ain't any plain PNGs / JPGs embedded. They are [SVGs](https://www.w3schools.com/graphics/svg_examples.asp) & there are dedicated people on the team who work on writing those SVG roadmaps. 

In that context, your's prepared roadmap images can't be directly pushed to the site, instead they will be used as a guide to create the actual one (in SVG format). Secondly if you want to contribute to the roadmaps in some other way, you can request PR [here](https://github.com/commclassroom/roadmaps).

**Comment by shoaibshebi** [comment]
@cosmicqbit Yes the roadmaps are not PNGs that is why I mentioned it's rough sketch, that I will propose.
Thanks you helped

**Comment by shoaibshebi** [comment]
@kaiwalyakoparkar 

Please find the PR against this id #153 
I have added the instructions in the PR and also in the README-PRIVATE-BC.md file

Happy to help 😊
Thanks

**Comment by shoaibshebi** [comment]
@kaiwalyakoparkar 

Please use this link to view the pdf file 
https://drive.google.com/file/d/1zwqoeQKHZ-Nw4_XZRpzOSW7hFWc_6K6G/view?usp=sharing

Thanks

**Reply by kaiwalyakoparkar** [reply]
@siddhant-khisty can you please start working on additing this to our current roadmap?

**Reply by siddhant-khisty** [reply]
will do
